components/Promotion/basic_usage.rst

```diff
@@ -0,0 +1,323 @@
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getPromotions()
+        {
+            return $this->promotions;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function hasPromotion(PromotionInterface $promotion)
+        {
+            return $this->promotions->contains($promotion);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getPromotionSubjectTotal()
+        {
+            //implementation
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function addPromotion(PromotionInterface $promotion)
+        {
+            if (!$this->hasPromotion($promotion)) {
+                $this->promotions->add($promotion);
+            }
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function removePromotion(PromotionInterface $promotion)
+        {
+            if($this->hasPromotion($promotion))
+            {
+                $this->promotions->removeElement($promotion);
+            }
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getPromotionSubjectCount()
+        {
+            return $this->getQuantity();
+        }
+
+        /**
+         * @return int
+         */
+        public function getUnitPrice()
+        {
+            return $this->unitPrice;
+        }
+
+        /**
+         * @param int $price
+         */
+        public function setUnitPrice($price)
+        {
+            $this->unitPrice = $price;
+        }
+
+        /**
+         * @return int
+         */
+        public function getTotal()
+        {
+            return $this->getUnitPrice() * $this->getQuantity();
+        }
+    }
+
+.. _component_promotion_generator_coupon-generator:
+
+CouponGenerator
+---------------
+
+In order to automate the process of coupon generation the component provides us with a Coupon Generator.
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Promotion\Model\Promotion;
+    use Sylius\Component\Promotion\Generator\Instruction;
+    use Sylius\Component\Promotion\Generator\CouponGenerator;
+
+    $promotion = new Promotion();
+
+    $instruction = new Instruction(); // $amount = 5 by default
+
+    /**
+     * @param RepositoryInterface    $repository
+     * @param EntityManagerInterface $manager
+     */
+    $generator = new CouponGenerator($repository, $manager);
+
+    //This will generate and persist 5 coupons into the database
+    //basing on the instruction provided for the given promotion object
+    $generator->generate($promotion, $instruction);
+
+    // We can also generate one unique code, and assign it to a new Coupon.
+    $code = $generator->generateUniqueCode();
+    $coupon = new Coupon();
+    $coupon->setCode($code);
+
+Checkers
+--------
+
+.. _component_promotion_checker_item-count-rule-checker:
+
+ItemCountRuleChecker
+~~~~~~~~~~~~~~~~~~~~
```

Comment:
This is one of the checkers, which I'd move to separate file with all other checkers. The **PromotionEligibilityChecker** is the main service, which should be here on top. :)